packages/instrumentor/instrument.ts

```diff
@@ -114,16 +135,29 @@ function instrument(
 	code: string,
 	filename: string,
 	shouldInstrument: FilePredicate,
-	shouldHook: FilePredicate
+	shouldHook: FilePredicate,
+	idStrategy: EdgeIdStrategy
 ) {
 	const transformations: PluginItem[] = [];
-	if (shouldInstrument(filename)) {
-		transformations.push(codeCoverage, compareHooks);
+	const shouldInstrumentFile = shouldInstrument(filename);
+	if (shouldInstrumentFile) {
```

Comment:
This function gets more and more complex. Nothing to do for now, but we should start thinking about better abstractions to add plugins without needing to extend the function further.

Reply:
Agreed. It will become even more complex with the coverage PR.